pkg/storegateway/bucket.go

```diff
@@ -1717,12 +1719,13 @@ func (b *bucketBlock) readIndexRange(ctx context.Context, off, length int64) ([]
 	return buf.Bytes(), nil
 }
 
-func (b *bucketBlock) readChunkRange(ctx context.Context, seq int, off, length int64, chunkRanges byteRanges) (*[]byte, error) {
+func (b *bucketBlock) readChunkRange(ctx context.Context, seq int, off, length int64, chunkRanges byteRanges, chunkSlabs *pool.SafeSlabPool[byte]) (*[]byte, error) {
```

Comment:
[nit] I would name the new param `chunksPool` instead of `chunkSlabs` to keep consistency with the rest of the code. Really a nit!